File: app/src/main/java/freddy/fudi/callbacks/AbstractCallback.java
package freddy.fudi.callbacks;

/**
 * Created by romelldominguez on 9/19/16.
 */
public abstract class AbstractCallback implements Callback {

    public abstract void successed(Object... objects);
    public abstract void failed(Exception exception, Object... objects);

    @Override
    public void failed(Object... objects) {
        failed(null, objects);
    }
}

File: app-kotlin/src/main/java/com/romellfudi/callbacks/AbstractCallback.kt
package com.romellfudi.callbacks

/**
 * Created by romelldominguez on 9/19/16.
 */
abstract class AbstractCallback : Callback {
    abstract override fun successed(vararg objects: Any)
    abstract fun failed(exception: Exception?, vararg objects: Any?)
    override fun failed(vararg objects: Any) {
        failed(null, *objects)
    }
}